### PR TITLE
MP Reactive Messaging first drop

### LINKED ISSUE #6685: Create feature mpReactiveMessaging-1.0
Create a public feature mpReactiveMessaging-1.0

### PR DESCRIPTION
**Pull request by tevans78**
#build 
fixes #6685

### CODE REVIEW

File: dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import io.smallrye.reactive.messaging.extension.StreamProducer;
+import io.smallrye.reactive.messaging.impl.ConfiguredStreamFactory;
+import io.smallrye.reactive.messaging.impl.InternalStreamRegistry;
+import io.smallrye.reactive.messaging.impl.StreamFactoryImpl;
+
+@Component(service = WebSphereCDIExtension.class, configurationPolicy = ConfigurationPolicy.IGNORE, property = { "service.vendor=IBM" })
+public class OLReactiveMessagingExtension extends ReactiveMessagingExtension implements Extension, WebSphereCDIExtension {
+
+    public OLReactiveMessagingExtension() {
+        //System.out.println("OLReactiveMessagingExtension.ctor");
```
Comment (hutchig) [inline dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java]: remove please

File: dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    void afterBeanDiscovery2(@Observes AfterBeanDiscovery discovery, BeanManager beanManager) {
+        //System.out.println("OLReactiveMessagingExtension.afterBeanDiscovery2");
```
Comment (hutchig) [inline dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java]: remove please

File: dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    void beforeBeanDiscovery(@Observes BeforeBeanDiscovery discovery, BeanManager beanManager) {
+        //System.out.println("OLReactiveMessagingExtension.beforeBeanDiscovery");
```
Comment (hutchig) [inline dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java]: remove please

File: dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
```
Comment (hutchig) [inline dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java]: This definately not legal to claim as ours. So we need to change it.

File: dev/com.ibm.ws.io.smallrye.reactive.messaging/src/io/smallrye/reactive/messaging/extension/ReactiveMessagingExtension.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
+package io.smallrye.reactive.messaging.extension;
```
Comment (hutchig) [inline dev/com.ibm.ws.io.smallrye.reactive.messaging/src/io/smallrye/reactive/messaging/extension/ReactiveMessagingExtension.java]: Apache 2 need - attribution and notices obligation observed if you create/ship a derivative work

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/simple/SimpleRxMessagingBean.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+@ApplicationScoped
+public class SimpleRxMessagingBean {
+
+    @Outgoing("source")
+    public PublisherBuilder<String> source() {
+        return ReactiveStreams.of("Length 8", "Length 9!", "Length 10!", "Length 11!!", "Length 12!!!");
+    }
+
+    @Incoming("source")
+    @Outgoing("uppercase")
```
Comment (hutchig) [inline dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/simple/SimpleRxMessagingBean.java]: uppercased would be better

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/simple/SimpleRxMessagingBean.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    public String toUpperCase(String payload) {
+        return payload.toUpperCase();
+    }
+
+    @Incoming("uppercase")
+    @Outgoing("filter")
```
Comment (hutchig) [inline dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/simple/SimpleRxMessagingBean.java]: filtered would be better

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/simple/SimpleRxMessagingServlet.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+@WebServlet("/")
+public class SimpleRxMessagingServlet extends FATServlet {
+
+    @Inject
+    private SimpleRxMessagingBean rxBean;
+
+    @Test
+    public void prod() {
+        assertEquals("PRODDED!", rxBean.prod());
```
Comment (hutchig) [inline dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/simple/SimpleRxMessagingServlet.java]: This needs to be a better name if the FAT bucket is not experimental.

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/suite/SimpleTest.java [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.stopServer();
+    }
+
+    @Test
+    public void simpleRXMessagingTest() throws Exception {
+        //TODO this test needs improving
+
+        List<String> eight = server.findStringsInLogs("LENGTH 8");
```
Comment (hutchig) [inline dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/suite/SimpleTest.java]: Is there no timing coordingation of if the test has finished running or not before the data is checked? If think it is best to make this experimantal (discussed/agreed with Tom) until we know this is reliable as we don't want to trip up builds. The problem is that the Messaging streams 'just run' when the app is dropped in.

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/test-applications/CDIFaultTolerance.war/resources/META-INF/microprofile-config.properties [hunk 10]
```diff
@@ -0,0 +1,25 @@
+###############################################################################
```
Comment (hutchig) [inline dev/com.ibm.ws.microprofile.reactive.messaging_fat/test-applications/CDIFaultTolerance.war/resources/META-INF/microprofile-config.properties]: Did you mean to change this?

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/test-applications/CDIFaultTolerance.war/resources/META-INF/permissions.xml [hunk 11]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
```
Comment (hutchig) [inline dev/com.ibm.ws.microprofile.reactive.messaging_fat/test-applications/CDIFaultTolerance.war/resources/META-INF/permissions.xml]: Changed?

File: dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java (hunk 1 above)
Comment (tevans78) [reply dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java]: done

File: dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java (hunk 3 above)
Comment (tevans78) [reply dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java]: done

File: dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java (hunk 2 above)
Comment (tevans78) [reply dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java]: done

File: dev/com.ibm.ws.io.smallrye.reactive.messaging/src/io/smallrye/reactive/messaging/extension/ReactiveMessagingExtension.java (hunk 5 above)
Comment (tevans78) [reply dev/com.ibm.ws.io.smallrye.reactive.messaging/src/io/smallrye/reactive/messaging/extension/ReactiveMessagingExtension.java]: done

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/simple/SimpleRxMessagingBean.java (hunk 6 above)
Comment (tevans78) [reply dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/simple/SimpleRxMessagingBean.java]: done

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/simple/SimpleRxMessagingBean.java (hunk 7 above)
Comment (tevans78) [reply dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/simple/SimpleRxMessagingBean.java]: done

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/simple/SimpleRxMessagingServlet.java (hunk 8 above)
Comment (tevans78) [reply dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/simple/SimpleRxMessagingServlet.java]: done ... rewrote the test a bit

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/suite/SimpleTest.java (hunk 9 above)
Comment (tevans78) [reply dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/suite/SimpleTest.java]: i have added some coordination ... but i have not made it experimental; We want to know if it fails in a random build on a random platform. If it fails we need to fix it.

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/test-applications/CDIFaultTolerance.war/resources/META-INF/microprofile-config.properties (hunk 10 above)
Comment (tevans78) [reply dev/com.ibm.ws.microprofile.reactive.messaging_fat/test-applications/CDIFaultTolerance.war/resources/META-INF/microprofile-config.properties]: nope

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/test-applications/CDIFaultTolerance.war/resources/META-INF/permissions.xml (hunk 11 above)
Comment (tevans78) [reply dev/com.ibm.ws.microprofile.reactive.messaging_fat/test-applications/CDIFaultTolerance.war/resources/META-INF/permissions.xml]: didn't mean to

**Comment by tevans78** [comment]
Commits squashed.
#build

**Comment by skasund** [comment]
No messages for L2 to review

**Comment by tevans78** [comment]
no idea why the PB failed, running again
#build

**Comment by tevans78** [comment]
PB Good. Merging

**Pull request merged**